Mathlib/Analysis/InnerProductSpace/Simultaneous.lean

```diff
@@ -0,0 +1,108 @@
+/-
+Copyright (c) 2024 Jon Bannon. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Jon Bannon, Jack Cheverton, Samyak Dhar Tuladhar
+-/
+
+import Mathlib.Analysis.InnerProductSpace.Spectrum
+import Mathlib.Analysis.InnerProductSpace.Projection
+import Mathlib.Order.CompleteLattice
+import Mathlib.LinearAlgebra.Eigenspace.Basic
+
+/-! # Simultaneous eigenspaces of a commuting pair of symmetric operators
```

Comment:
"Simultaneous", while I've heard the term used before, I think only makes sense in the context of *diagonalization*, and, strictly speaking, there is no diagonalization here. (in the sense that we don't get a collection of eigenvalues and eigenvectors.). Also, don't focus on pairs here, since you want to add the version for finite collections of commuting operators shortly.
```suggestion
/-! # Joint eigenspaces of commuting symmetric operators
```